agent/installer/internal/algo/mock_ubuntu_with_error.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package algo
+
+import "errors"
+
+type mockStep struct {
+	Step
+	Desc    string
+	Err     error
+	doCnt   *int
+	undoCnt *int
+	*BaseK8sInstaller
+}
+
+func (s *mockStep) do() error {
+	s.OutputBuilder.Msg("Installing: " + s.Desc)
+	*s.doCnt++
+	return s.Err
+}
+
+func (s *mockStep) undo() error {
+	s.OutputBuilder.Msg("Uninstalling: " + s.Desc)
+	*s.undoCnt++
+	return s.Err
+}
+
+type MockUbuntuWithError struct {
+	BaseK8sInstaller
+	errorOnStep int
+	curStep     int
+	doCnt       int
```

Comment:
Here you can have *MockUbuntuWithError and access these thru it.
Counters don't tell us if we undid the same steps or the others.
You can also collect the names of do and undo steps. Then assert they match (undo is in reverse order).